elytron/src/main/java/org/wildfly/extension/elytron/AuditResourceDefinitions.java

```diff
@@ -510,7 +510,9 @@ public void validateParameter(String parameterName, ModelNode value) throws Oper
             if (value.isDefined()) {
                 final String suffix = value.asString();
                 try {
-                    if (denySeconds && (suffix.contains("s") || suffix.contains("S"))) {
+                    //remove text quoted using single quotes
+                    String suffixToCheck = suffix.replaceAll("'([^']*?)'", "");
+                    if (denySeconds && (suffixToCheck.contains("s") || suffixToCheck.contains("S"))) {
```

Comment:
Maybe not a huge deal as this is likely not invoked all that often but something like the following would likely perform a bit better.
```java
for (int i = 0; i < suffix.length(); i++) {
    char c = suffix.charAt(i);
    if (c == '\'') {
        c = suffix.charAt(++i);
        while (c != '\'') {
            c = suffix.charAt(++i);
        }
    }
    if (c == 's' || c == 'S') {
        throw ElytronSubsystemMessages.ROOT_LOGGER.suffixContainsMillis(suffix);
    }
}
```

Either way the `denySeconds` should probably be checked before we use regex on the string.

This is going to have to change in the other validators, the logging subsystem and management audit logging, as well. I'm okay with fixing the other two in a separate PR or making the issue more generic and fixing them all in the same PR.

Reply:
@jamezp Thanks for the review. I'll make the changes, create new issues and fix the other validators in separate PR.